terraform/modules/cloud-build-docker/build_image.py

```diff
@@ -30,25 +31,31 @@ def run_command(cmd, **kwargs):
 def get_image_digest(image_uri, tag, project_id):
     """Query the digest of an existing image."""
     try:
+        # Use list-tags to get all tags, then filter in Python for exact match
+        # This avoids issues where "tags:latest" matches both "latest" and "latest-builder"
```

Comment:
If the issue is you want an exact match, use `tags={tag}` instead of `tags:{tag}` on (the old) line 41.  At least, that's my reading of `gcloud topic filters`.